### PR TITLE
feat: [SSCA-345] routes for new pages, route utils

### PR DESCRIPTION
**Pull request by udayvunnam**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/utils/routeUtils.ts [hunk 1]
```diff
@@ -193,6 +194,23 @@ export function withProjectIdentifier<T>(fn: (args: T) => string) {
...
+    const path = fn(params)
+
+    const { accountId, orgIdentifier, projectIdentifier, module = 'home' } = params
+
+    return `/account/${accountId}/${module}/orgs/${orgIdentifier}/projects/${projectIdentifier}/${path.replace(
+      /^\//,
+      ''
+    )}`
+  }
+}
```
Comment (udayvunnam) [inline src/modules/10-common/utils/routeUtils.ts:211]: This is duplicated at several places, will update the usage in a different PR

File: src/modules/10-common/RouteDefinitions.ts [hunk 2]
```diff
@@ -2282,7 +2282,8 @@ const routes = {
   // SSCA
   toSSCA: withAccountId(() => '/ssca'),
   toSSCAOverview: withAccountId(() => '/ssca/overview'),
-  toSSCAGettingStarted: withAccountId(() => '/ssca/getting-started'),
+  toSscaArtifacts: withProjectLevelRoute(() => '/artifacts'),
```
Comment (pankajsingh-harness) [inline src/modules/10-common/RouteDefinitions.ts:2285]:
```suggestion
  toSSCAArtifacts: withProjectLevelRoute(() => '/artifacts'),
```

File: src/modules/10-common/RouteDefinitions.ts [hunk 3]
```diff
@@ -2282,7 +2282,8 @@ const routes = {
...
+  toSscaComponents: withProjectLevelRoute(() => '/components'),
```
Comment (pankajsingh-harness) [inline src/modules/10-common/RouteDefinitions.ts:2286]:
```suggestion
  toSSCAComponents: withProjectLevelRoute(() => '/components'),
```

**Review by pankajsingh-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good Uday

File: src/modules/27-platform/default-settings/RouteDestinations.tsx [hunk 4]
```diff
@@ -42,27 +42,27 @@ import {
 DefaultSettingsFactory.registerCategory('CORE', {
   icon: 'cog',
   label: 'common.settingCategory.general',
-  modulesWhereCategoryWillBeDisplayed: ['cd', 'ce', 'cf', 'chaos', 'ci', 'cv', 'code', 'sto']
+  modulesWhereCategoryWillBeDisplayed: ['cd', 'ce', 'cf', 'chaos', 'ci', 'cv', 'code', 'sto', 'ssca']
```
Comment (ishantmehtaharness) [inline src/modules/27-platform/default-settings/RouteDestinations.tsx:45]: Can you please share screenshot ofwhat it does

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**